src/Service/Services/ResolverTypeInterceptor.cs

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Azure.DataApiBuilder.Service.Resolvers;
+using Azure.DataApiBuilder.Service.Services;
+using HotChocolate.Configuration;
+using HotChocolate.Resolvers;
+using HotChocolate.Types;
+using HotChocolate.Types.Descriptors.Definitions;
+
+internal sealed class ResolverTypeInterceptor : TypeInterceptor
+{
+    private readonly FieldMiddlewareDefinition _queryMiddleware;
+    private readonly FieldMiddlewareDefinition _mutationMiddleware;
+    private readonly PureFieldDelegate _leafFieldResolver;
+    private readonly PureFieldDelegate _objectFieldResolver;
+    private readonly PureFieldDelegate _listFieldResolver;
+    
+    public ResolverTypeInterceptor(ExecutionHelper executionHelper)
+    {
+        _queryMiddleware = 
+            new FieldMiddlewareDefinition(
+                next => async context =>
+                {
+                    await executionHelper.ExecuteMutateAsync(context).ConfigureAwait(false);
+                    await next(context).ConfigureAwait(false);
+                });
+        
+        _mutationMiddleware = 
+            new FieldMiddlewareDefinition(
+                next => async context =>
+                {
+                    await executionHelper.ExecuteMutateAsync(context).ConfigureAwait(false);
+                    await next(context).ConfigureAwait(false);
+                });
+
+        _leafFieldResolver = ctx => ExecutionHelper.ExecuteLeafField(ctx);
+        _objectFieldResolver = ctx => executionHelper.ExecuteObjectField(ctx);
+        _listFieldResolver = ctx => executionHelper.ExecuteListField(ctx);
+    }
+    
+    public override void OnBeforeCompleteType(
+        ITypeCompletionContext completionContext,
+        DefinitionBase? definition,
+        IDictionary<string, object?> contextData)
+    {
+        // We are only interested in object types here as only object types can have resolvers.
+        if (definition is not ObjectTypeDefinition objectTypeDef)
+        {
+            return;
+        }
+
+        if (completionContext.IsQueryType ?? false)
+        {
+            foreach (ObjectFieldDefinition field in objectTypeDef.Fields)
+            {
+                field.MiddlewareDefinitions.Add(_mutationMiddleware);
```

Comment:
Did addition of the middleware definitions got switched up here - as in should this be _queryMiddleware if completionContext.IsQueryType?

Reply:
yes